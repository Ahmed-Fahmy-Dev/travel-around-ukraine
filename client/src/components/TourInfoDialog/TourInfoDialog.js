/* eslint-disable react/jsx-props-no-spreading */
import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { styled, alpha } from '@mui/material/styles';
import MuiBox from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import { Button, MenuItem as MuiMenuItem, Select as MuiSelect, Stack } from '@mui/material';
// Icons
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import CalendarMonthIcon from '@mui/icons-material/CalendarMonth';
import AccessTimeIcon from '@mui/icons-material/AccessTime';
import FlightTakeoffIcon from '@mui/icons-material/FlightTakeoff';
import KeyboardReturnIcon from '@mui/icons-material/KeyboardReturn';

import formatTourDate from '../../utils/formatTourDate';

const currency = {
  eur: '€',
  usd: '$',
  uah: '₴',
};

const BoxWrapper = styled(MuiBox)(({ theme }) => ({
  position: 'sticky',
  top: 20,
  padding: '20px 20px 30px',
  border: `1px solid ${theme.palette.divider}`,
  borderRadius: 5,
}));

const Section = styled(MuiBox)(({ theme }) => ({
  borderBottom: `1px solid ${theme.palette.divider}`,
  paddingBottom: 15,
  marginBottom: 20,
}));

const Title = styled((props) => <Typography variant="h3" color="black" component="p" {...props} />)({
  textTransform: 'uppercase',
  fontWeight: 700,
  marginBottom: 20,
});

const Cost = styled(Typography)(({ theme }) => ({
  fontWeight: 700,
  marginBottom: 0,
  fontSize: 35,
  lineHeight: 1,
  color: theme.palette.primary.main,
}));

const Select = styled(MuiSelect)(({ theme }) => ({
  marginBottom: '10px',
  fontWeight: 700,
  '.MuiOutlinedInput-notchedOutline': {
    borderColor: `${theme.palette.divider}`,
  },
  '&:hover .MuiOutlinedInput-notchedOutline': {
    borderColor: `${theme.palette.primary.main}`,
  },
  '.MuiSelect-icon': {
    transition: 'transform .2s',
  },
}));

const MenuItem = styled(MuiMenuItem)(({ theme }) => ({
  '&:hover': {
    backgroundColor: alpha(theme.palette.primary.main, 0.15),
    color: theme.palette.primary.main,
  },

  '&.Mui-selected': {
    color: theme.palette.primary.main,
    fontWeight: 700,
  },
}));

const DetailsText = styled((props) => <Typography gutterBottom={false} {...props} />)({});

const TourInfoDialog = ({ included, cost, dates, details }) => {
  const [costCurrency, setCostCurrency] = useState('eur');

  return (
    <BoxWrapper>
      <Section>
        <Title>Included</Title>
        <Stack direction="row" justifyContent="space-between" flexWrap="wrap" gap="12px">
          {included?.map(({ icon, service }) => (
            <Stack key={service} direction="row" gap="5px" alignItems="center">
              {icon}
              <Typography gutterBottom={false}>{service}</Typography>
            </Stack>
          ))}
        </Stack>
      </Section>
      <Section>
        <Title>Dates</Title>
        <Stack direction="row" gap="5px" alignItems="center">
          <CalendarMonthIcon color="primary" />
          <Typography gutterBottom={false}>
            {formatTourDate(dates.beginDate)} - {formatTourDate(dates.endDate)}
          </Typography>
        </Stack>
      </Section>
      <Section>
        <Title>Details</Title>
        <Stack gap="5px">
          <Stack direction="row" alignItems="center" gap="5px">
            <AccessTimeIcon color="primary" />
            <DetailsText>Duration</DetailsText>
            <DetailsText marginLeft="auto">{details.duration}</DetailsText>
          </Stack>
          <Stack direction="row" alignItems="center" gap="5px">
            <FlightTakeoffIcon color="primary" />
            <DetailsText>Departs</DetailsText>
            <DetailsText marginLeft="auto">{details.departs}</DetailsText>
          </Stack>
          <Stack direction="row" alignItems="center" gap="5px">
            <KeyboardReturnIcon color="primary" />
            <DetailsText>Returns</DetailsText>
            <DetailsText marginLeft="auto">{details.returns}</DetailsText>
          </Stack>
        </Stack>
      </Section>
      <Section>
        <Title>Currency</Title>
        <Select
          fullWidth
          value={costCurrency}
          onChange={(e) => setCostCurrency(e.target.value)}
          IconComponent={ExpandMoreIcon}
        >
          {Object.keys(currency).map((key) => (
            <MenuItem key={key} value={key}>
              {key.toUpperCase()}
            </MenuItem>
          ))}
        </Select>
      </Section>
      <MuiBox>
        <Title>Total</Title>
        <Stack direction="row" justifyContent="space-between" alignItems="center">
          <Cost>
            {currency[costCurrency]}
            {cost[costCurrency]}
          </Cost>
          <Button variant="contained" sx={{ paddingInline: '30px' }} disableElevation>
            Add to Cart
          </Button>
        </Stack>
      </MuiBox>
    </BoxWrapper>
  );
};

TourInfoDialog.propTypes = {
  included: PropTypes.arrayOf(PropTypes.object).isRequired,
  cost: PropTypes.oneOfType([PropTypes.object, PropTypes.string]).isRequired,
  dates: PropTypes.objectOf(PropTypes.instanceOf(Date)).isRequired,
  details: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default TourInfoDialog;
