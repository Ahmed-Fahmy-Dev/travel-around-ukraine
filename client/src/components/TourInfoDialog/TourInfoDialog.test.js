import { render } from '@testing-library/react';
import TourInfoDialog from './TourInfoDialog';

const included = [
  { icon: '', service: 'Professional guide' },
  { icon: '', service: 'Accomodation' },
];

const cost = {
  eur: 70,
  usd: 70,
  uah: 2584,
};

const dates = { beginDate: new Date('2022-02-01'), endDate: new Date('2022-02-25') };

const details = {
  duration: '3 days',
  departs: 'First city',
  returns: 'Second city',
};

describe('TourInfoDialog snapshot testing', () => {
  test('should TourInfoDialog without close button match snapshot', () => {
    const { asFragment } = render(<TourInfoDialog included={included} cost={cost} dates={dates} details={details} />);

    expect(asFragment()).toMatchSnapshot();
  });
});
