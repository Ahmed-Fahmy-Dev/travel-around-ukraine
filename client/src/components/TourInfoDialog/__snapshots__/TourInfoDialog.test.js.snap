// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TourInfoDialog snapshot testing should TourInfoDialog without close button match snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-1l5oawv"
  >
    <div
      class="MuiBox-root css-1kwkjfv"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1jjn5p1-MuiTypography-root"
      >
        Included
      </h3>
      <div
        class="css-196xmg7-MuiStack-root"
      >
        <div
          class="css-1j887li-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Professional guide
          </p>
        </div>
        <div
          class="css-1j887li-MuiStack-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Accomodation
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1kwkjfv"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1jjn5p1-MuiTypography-root"
      >
        Dates
      </h3>
      <div
        class="css-1j887li-MuiStack-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
          data-testid="CalendarMonthIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M19 4h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V10h14v10zM9 14H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2zm-8 4H7v-2h2v2zm4 0h-2v-2h2v2zm4 0h-2v-2h2v2z"
          />
        </svg>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          2022.02.01 - 2022.02.25
        </p>
      </div>
    </div>
    <div
      class="MuiBox-root css-1kwkjfv"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1jjn5p1-MuiTypography-root"
      >
        Details
      </h3>
      <div
        class="css-14tdmau-MuiStack-root"
      >
        <div
          class="css-1ub4dmn-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="AccessTimeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M11.99 2C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
            />
            <path
              d="M12.5 7H11v6l5.25 3.15.75-1.23-4.5-2.67z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-16gwivt-MuiTypography-root"
          >
            Duration
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-32hrya-MuiTypography-root"
          >
            3 days
          </p>
        </div>
        <div
          class="css-1ub4dmn-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="FlightTakeoffIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M2.5 19h19v2h-19v-2zm19.57-9.36c-.21-.8-1.04-1.28-1.84-1.06L14.92 10l-6.9-6.43-1.93.51 4.14 7.17-4.97 1.33-1.97-1.54-1.45.39 2.59 4.49s7.12-1.9 16.57-4.43c.81-.23 1.28-1.05 1.07-1.85z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-16gwivt-MuiTypography-root"
          >
            Departs
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-32hrya-MuiTypography-root"
          >
            First city
          </p>
        </div>
        <div
          class="css-1ub4dmn-MuiStack-root"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium css-ci5apu-MuiSvgIcon-root"
            data-testid="KeyboardReturnIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 7v4H5.83l3.58-3.59L8 6l-6 6 6 6 1.41-1.41L5.83 13H21V7z"
            />
          </svg>
          <p
            class="MuiTypography-root MuiTypography-body1 css-16gwivt-MuiTypography-root"
          >
            Returns
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-32hrya-MuiTypography-root"
          >
            Second city
          </p>
        </div>
      </div>
    </div>
    <div
      class="MuiBox-root css-1kwkjfv"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1jjn5p1-MuiTypography-root"
      >
        Currency
      </h3>
      <div
        class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth css-rtvg6-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
          role="button"
          tabindex="0"
        >
          EUR
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
          tabindex="-1"
          value="eur"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
          data-testid="ExpandMoreIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
        <fieldset
          aria-hidden="true"
          class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
        >
          <legend
            class="css-ihdtdm"
          >
            <span
              class="notranslate"
            >
              ​
            </span>
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      class="MuiBox-root css-0"
    >
      <h3
        class="MuiTypography-root MuiTypography-h3 css-1jjn5p1-MuiTypography-root"
      >
        Total
      </h3>
      <div
        class="css-3y1137-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-1hoyjyq-MuiTypography-root"
        >
          €70
        </p>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButton-disableElevation css-fwdzms-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          Add to Cart
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
